applications/asset_tracker_v2/src/cloud/nrf_cloud_integration.c

```diff
@@ -0,0 +1,366 @@
+#include "cloud/cloud_wrapper.h"
+#include <zephyr.h>
+#include <net/nrf_cloud.h>
+#include <net/mqtt.h>
+
+#include "cJSON.h"
+#include "json_helpers.h"
+#include "json_protocol_names.h"
+
+#define MODULE nrf_cloud_integration
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_CLOUD_INTEGRATION_LOG_LEVEL);
+
+#define NRF_CLOUD_SERVICE_INFO "{"								\
+					"\"state\":{"						\
+						"\"reported\":{"				\
+							"\"device\":{"				\
+								"\"serviceInfo\":{"		\
+									"\"ui\":["		\
+										"\"GPS\","	\
+										"\"HUMID\","	\
+										"\"RSRP\","	\
+										"\"BUTTON\","	\
+										"\"TEMP\""	\
+									"],"			\
+									"\"fota_v2\":["		\
+										"\"APP\","	\
+										"\"MODEM\","	\
+										"\"BOOT\""	\
+									"]"			\
+								"}"				\
+							"}"					\
+						"}"						\
+					"}"							\
+				"}"
+
+#define REQUEST_DEVICE_STATE_STRING ""
+
+static cloud_wrap_evt_handler_t wrapper_evt_handler;
+
+static void cloud_wrapper_notify_event(const struct cloud_wrap_event *evt)
+{
+	if ((wrapper_evt_handler != NULL) && (evt != NULL)) {
+		wrapper_evt_handler(evt);
+	} else {
+		LOG_ERR("Library event handler not registered, or empty event");
+	}
+}
+
+static int send_service_info(void)
+{
+	int err;
+
+	struct nrf_cloud_tx_data msg = {
+		.data.ptr = NRF_CLOUD_SERVICE_INFO,
+		.data.len = sizeof(NRF_CLOUD_SERVICE_INFO) - 1,
+		.qos = MQTT_QOS_0_AT_MOST_ONCE,
+		.topic_type = NRF_CLOUD_TOPIC_STATE,
+	};
+
+	err = nrf_cloud_send(&msg);
+	if (err) {
+		LOG_ERR("nrf_cloud_send, error: %d", err);
+		return err;
+	}
+
+	LOG_DBG("nRF Cloud service info sent: %s", NRF_CLOUD_SERVICE_INFO);
+
+	return 0;
+}
+
+static void nrf_cloud_event_handler(const struct nrf_cloud_evt *evt)
+{
+	struct cloud_wrap_event cloud_wrap_evt = { 0 };
+	bool notify = false;
+
+	switch (evt->type) {
+	case NRF_CLOUD_EVT_TRANSPORT_CONNECTING:
+		LOG_DBG("NRF_CLOUD_EVT_TRANSPORT_CONNECTING");
+		break;
+	case NRF_CLOUD_EVT_TRANSPORT_CONNECTED:
+		LOG_DBG("NRF_CLOUD_EVT_TRANSPORT_CONNECTED");
+		break;
+	case NRF_CLOUD_EVT_READY:
+		LOG_DBG("NRF_CLOUD_EVT_READY");
+		cloud_wrap_evt.type = CLOUD_WRAP_EVT_CONNECTED;
+		notify = true;
+
+		int err = send_service_info();
```

Comment:
Wonder why the compiler allows this without wrapping the case in curly braces

Reply:
Beats me...